src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureNormalizers.java

```diff
@@ -0,0 +1,197 @@
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class StoredFeatureNormalizers {
+
+
+    public enum Type implements Writeable {
+        STANDARD,
+        MIN_MAX;
+
+
+        public static Type readFromStream(StreamInput in) throws IOException {
+            int ord = in.readVInt();
+            for (Type type: Type.values()) {
+                if (type.ordinal() == ord) {
+                    return type;
+                }
+            }
+            throw new ElasticsearchException("unknown normalizer type during serialization [" + ord + "]");
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVInt(this.ordinal());
+        }
+    }
+
+    public static final ObjectParser.NamedObjectParser<FeatureNormDefinition, Void> PARSER;
+    private static final ParseField STANDARD = new ParseField("standard");
+    private static final ParseField MIN_MAX = new ParseField("min_max");
+
+    static {
+
+        PARSER = (XContentParser p, Void c, String featureName) -> {
+            // this seems really intended for switching on the key (here featureName) and making
+            // a decision, when in reality, we want to look a layer deeper and switch on that
+            ObjectParser<FeatureNormConsumer, Void> parser = new ObjectParser<>("feature_normalizers",
+                                                                                  FeatureNormConsumer::new);
+
+            BiConsumer<FeatureNormConsumer, StandardFeatureNormDefinition> setStd = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            BiConsumer<FeatureNormConsumer, MinMaxFeatureNormDefinition> setMinMax = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            parser.declareObject(setStd, StandardFeatureNormDefinition.PARSER, STANDARD);
+            parser.declareObject(setMinMax, MinMaxFeatureNormDefinition.PARSER, MIN_MAX);
+
+            FeatureNormConsumer parsedNorm  = parser.parse(p, c);
+            return parsedNorm.ftrNormDefn;
+
+        };
+
+    }
+
+    // A temp holder for parsing out of xcontent
+    private static class FeatureNormConsumer {
+        String featureName;
+        FeatureNormDefinition ftrNormDefn;
+
+        FeatureNormConsumer() {
+        }
+
+        public FeatureNormDefinition getFtrNormDefn() {
+            return this.ftrNormDefn;
+        }
+
+        public void setFtrNormDefn(FeatureNormDefinition ftrNormDefn) {
+            this.ftrNormDefn = ftrNormDefn;
```

Comment:
should prevent setting the normalizer twice by throwing IllegalArgumentException if this.ftrNormDefn is already set?